state_machine/applications/system-check/tests/nvm_access_test.py

```diff
@@ -0,0 +1,155 @@
+from print_utils import bold, normal
+
+
+def get_nvm_flags():
+    """
+    Find all of the bitFlags in the
+    _Satellite class.
+    """
+
+    sat_vars = vars(_Satellite)
+    nvm_flags = []
+    for v in sat_vars:
+        if type(sat_vars[v]) == bitFlag:
+            nvm_flags.append(v)
+
+    return nvm_flags
+
+def get_nvm_counters():
+    """
+    Find all of the multiBitFlag and multiByte counters in the
+    _Satellite class.
+    """
+
+    sat_vars = vars(_Satellite)
+    nvm_counters = []
+    for v in sat_vars:
+        if type(sat_vars[v]) == multiBitFlag or type(sat_vars[v]) == multiByte:
+            nvm_counters.append(v)
+
+    return nvm_counters
+
+def verify_value(field_str, value):
+    setattr(cubesat, field_str, value)
+    if not getattr(cubesat, field_str) == value:
+        return False
+    return True
+
+def verify_counter(counter_str):
+    """ verify that the counter values can be changed and stay in range """
+
+    # save current counter value
+    counter_prev = getattr(cubesat, counter_str)
+    maxval = vars(_Satellite)[counter_str].maxval
+
+    success = True
+    success &= verify_value(counter_str, 0)
+    success &= verify_value(counter_str, maxval)
+    success &= not verify_value(counter_str, maxval + 1)
+
+    # restore counter value
+    setattr(cubesat, counter_str, counter_prev)
+
+    return success
+
+def verify_flag(flag_str):
+    """ verify that the flag can be set to either 0 or 1 """
+
+    # save current flag value
+    flag_prev = getattr(cubesat, flag_str)
+
+    success = True
+    success &= verify_value(flag_str, 0)
+    success &= verify_value(flag_str, 1)
+
+    # restore flag value
+    setattr(cubesat, flag_str, flag_prev)
+
+    return success
+
+def test_counter_interference(result_dict):
+    """ check that updating any particular nvm counter doesn't affect
+    other parts of non volatile memory """
+
+    nvm_counters = get_nvm_counters()
+
+    # save original values of the counters
+    original_values = [getattr(cubesat, counter_str) for counter_str in nvm_counters]
+
+    # set each counter in nvm to their maximum values
+    for counter_str in nvm_counters:
+        maxval = vars(_Satellite)[counter_str].maxval
+        setattr(cubesat, counter_str, maxval)
+
+    # change each counter to 0, make sure no other counter changed
+    success = True
+    interfering_counters = []
+    for counter_str in nvm_counters:
+        setattr(cubesat, counter_str, 0)
+
+        # check that the remaining counter vals are the same
+        for remaining_counter_str in nvm_counters:
+            if remaining_counter_str == counter_str:
+                continue
+
+            maxval = vars(_Satellite)[remaining_counter_str].maxval
+            val = getattr(cubesat, remaining_counter_str)
+            if not val == maxval:
+                success = False
+                interfering_counters.append(f"{counter_str} interfers with {remaining_counter_str}")
```

Comment:
This can result in duplicate outputs:
```
Failed Tests:
  - NVM_Counter_Interference: Inteferences: ['c_boot interfers with c_logfail', 'c_logfail interferes with c_boot']
 ```

Reply:
I think it's possible that interference would happen in one direction but not the other, so I'm going to keep this.